astroquery/esa/jwst/core.py

```diff
@@ -994,10 +1003,19 @@ def get_obs_products(self, *, observation_id=None, cal_level="ALL",
                                                 max_cal_level=max_cal_level,
                                                 is_url=True)
         params_dict['planeid'] = plane_ids
+
+        if self.__check_list_strings(product_type):
+            if type(product_type) is list:
+                tap_product_type = ",".join(str(elem) for elem in product_type)
+            else:
+                tap_product_type = product_type
+        else:
+            tap_product_type = None
```

Comment:
this function is only used in this one place, so I would instead suggest something in-place here along the line of the following few line. Arguably this doesn't gracefully handle the case when the input list is not full of strings, but the docs calls for strings, so duct-typeing should be sufficient. And with this the extra private function is not needed either.

```suggestion

        if isinstance(product_type, list):
            tap_product_type = ",".join(str(elem) for elem in product_type)
        elif isinstance(product_type, str):
            tap_product_type = product_type
        else:
            tap_product_type = None
```

Reply:
Brigitta, thank you very much for your comments and patience!
I was not using the --force in the push, my fault :)
About this method "check_list_strings" is used as in other submodules (ex. hubble), but you’re right, in jwst no, I can delete it and check the entries directly in "get_obs_products".
Cheers,
David